src/providers/eth_provider/blocks.rs

```diff
@@ -12,6 +12,8 @@ use mongodb::bson::doc;
 use reth_primitives::{BlockId, BlockNumberOrTag};
 use tracing::Instrument;
 
+pub type ExtendedTransaction = WithOtherFields<Transaction>;
+pub type ExtendedBlock = WithOtherFields<Block<ExtendedTransaction>>;
```

Comment:
This is exactly the goal of this PR, can you just move these constants to the appropriate files with header documentation each time to understand what these types define:
- `ExtendedTransaction` could be defined in the file where `StoredTransaction` is defined for example.
- `ExtendedBlock` could be defined in the file where `StoredHeader` is defined for example.

Then you can apply the changed to the whole codebase.

Reply:
When you say "defined" do you mean to add for example `ExtendedTransaction` to this file in `src/providers/eth_provider/database/types/transaction.rs`? in `types/ethereum.rs` we are using `StoredTransaction` and `StoredHeader` maybe that file? I'm not getting the "defined" part.

Reply:
Defined just means moving `pub type ExtendedTransaction = WithOtherFields<Transaction>;` to this file :)

Reply:
So moving `ExtendedTransaction` to this path `src/providers/eth_provider/database/types/transaction.rs`? And then importing the type where is neccessary.

Reply:
Yes correct it makes more sense for this type to be defined in appropriate files